src/dotnet/Chat.Service/ReactionsBackend.cs

```diff
@@ -0,0 +1,88 @@
+using ActualChat.Chat.Db;
+using Microsoft.EntityFrameworkCore;
+using Stl.Fusion.EntityFramework;
+
+namespace ActualChat.Chat;
+
+internal class ReactionsBackend : DbServiceBase<ChatDbContext>, IReactionsBackend
+{
+    private static readonly TileStack<long> IdTileStack = Constants.Chat.IdTileStack;
+    private IChatsBackend ChatsBackend { get; }
+
+    public ReactionsBackend(IServiceProvider services, IChatsBackend chatsBackend) : base(services)
+        => ChatsBackend = chatsBackend;
+
+    // [ComputeMethod]
+    public virtual async Task<ImmutableArray<ReactionSummary>> List(
+        string chatId,
+        long entryId,
+        CancellationToken cancellationToken)
+    {
+        var dbContext = CreateDbContext();
+        await using var _ = dbContext.ConfigureAwait(false);
+
+        var dbReactions = await dbContext.Reactions
+            .Where(x => x.ChatId == chatId && x.EntryId == entryId)
+            .ToListAsync(cancellationToken)
+            .ConfigureAwait(false);
+
+        return dbReactions.Select(x => x.ToModel())
+            .GroupBy(x => x.Emoji, StringComparer.Ordinal)
+            .Select(x => new ReactionSummary(x.Key, x.Count()))
+            .ToImmutableArray();
+    }
+
+    // [CommandHandler]
+    public virtual async Task React(IReactionsBackend.ReactCommand command, CancellationToken cancellationToken)
+    {
+        var reaction = command.Reaction;
+        var chatId = reaction.ChatId;
+        var entryId = reaction.EntryId;
+        if (Computed.IsInvalidating()) {
+            _ = List(chatId, entryId, default);
+            return;
+        }
+
+        var dbContext = await CreateCommandDbContext(cancellationToken).ConfigureAwait(false);
+        await using var __ = dbContext.ConfigureAwait(false);
+
+        var id = DbReaction.ComposeId(chatId, entryId, reaction.AuthorId);
```

Comment:
Why do you need `AuthorId` at `DbReaction`' CompositeId?

Reply:
otherwise reaction of user1 and user2 will have same id